servicetalk-loadbalancer/src/main/java/io/servicetalk/loadbalancer/RoundRobinLoadBalancer.java

```diff
@@ -829,7 +829,9 @@ boolean addConnection(C connection) {
                 Object[] newList = Arrays.copyOf(existing, existing.length + 1);
                 newList[existing.length] = connection;
 
-                Object newState = ActiveState.class.equals(previous.state.getClass()) ?
+                // If we were able to add a new connection to the list, we should mark the host as ACTIVE again and
+                // reset its failures counter.
+                Object newState = ActiveState.class.equals(previous.state.getClass()) || Host.isUnhealthy(previous) ?
```

Comment:
(general improvement suggestion) `ActiveState.class.equals..` -> consider adding a utility method for this too (it is done in a few spots). Also `isActiveAndHealthy` could call the new static method to share code too.